crates/eth-rpc/src/rpc.rs

```diff
@@ -0,0 +1,42 @@
+use jsonrpsee::RpcModule;
+use kakarot_rpc_core::client::config::{JsonRpcClientBuilder, StarknetConfig};
+use kakarot_rpc_core::client::KakarotClient;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+
+use crate::api::eth_api::EthApiServer;
+use crate::api::kakarot_custom_api::KakarotCustomApiServer;
+use crate::servers::eth_rpc;
+use crate::servers::eth_rpc::KakarotEthRpc;
+use crate::servers::kakarot_custom_rpc::{self, KakarotCustomRpc};
+
+pub struct KakarotRpc {
+    eth_rpc_module: RpcModule<KakarotEthRpc<JsonRpcClient<HttpTransport>>>,
+    kakarot_custom_rpc_module: RpcModule<KakarotCustomRpc<JsonRpcClient<HttpTransport>>>,
+}
+
+impl KakarotRpc {
+    pub fn new(starknet_config: StarknetConfig) -> Self {
+        let eth_rpc_module =
+            eth_rpc::KakarotEthRpc::new(Box::new(KakarotRpc::get_kakarot_client(starknet_config.clone()))).into_rpc();
+        let kakarot_custom_rpc_module =
+            kakarot_custom_rpc::KakarotCustomRpc::new(Box::new(KakarotRpc::get_kakarot_client(starknet_config)))
+                .into_rpc();
+
+        Self { eth_rpc_module, kakarot_custom_rpc_module }
+    }
+
+    pub fn get_rpc_module(&self) -> RpcModule<()> {
+        let mut rpc_module = RpcModule::new(());
```

Comment:
is there a slightly cleaner way to do this merging?

Reply:
Did this refactor which is cleaner, lemme know what your thoughts are.

Took some [inspiration](https://github.com/paradigmxyz/reth/blob/dbafe23cce45917a43f6640d71a0216fe3268428/crates/rpc/rpc-builder/src/lib.rs#L632) from reth, tried to keep our implementation as simple as possible.

<img width="866" alt="image" src="https://github.com/kkrt-labs/kakarot-rpc/assets/41180869/b062c5d2-0da0-4f7f-bf48-8dfd2046bddc">